### PR TITLE
[doc] publish the api policy

### PR DESCRIPTION
**Pull request by can-anyscale**
Publish the Ray API policy. 

There are things to improve, but with the combination from the code-completion from @peytondmurray and local build from @khluu, I have a much better developer experience.

Screenshot 2024-08-08 at 10 23 19 AM

Test:
- CI

### CODE REVIEW

File: doc/source/ray-contribute/stability.rst [hunk 1]
```diff
@@ -1,6 +1,6 @@
 .. _api-stability:
 
-API stability
+API Stability
```
Comment (can-anyscale) [inline doc/source/ray-contribute/stability.rst:3]: change this for consistency while i'm here

File: doc/source/ray-contribute/api-policy.rst [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    :widths: 20 16 16 16 16 16
+    :header-rows: 1
+
+    * - Policy/Exposure Level
+      - Stable Public API
+      - Beta Public API
+      - Alpha Public API
+      - Deprecated API
+      - Developer API
+    * - Must this API be documented?
```
[the remark this answers is not in the document]
Comment (angelinalg) [reply doc/source/ray-contribute/api-policy.rst:38]: Is it called an annotation or a decorator?

File: doc/source/ray-contribute/api-policy.rst [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+      - Yes
+      - Yes
+      - Up to the developers
+    * - Must a method API be annotated?
+      - Yes
+      - Yes
+      - Yes
+      - Yes
+      - No annotations mean it is a developer API by default
+    * - Must this API have an end-of-life date declared?
```
[the remark this answers is not in the document]
Comment (angelinalg) [reply doc/source/ray-contribute/api-policy.rst:50]: How do you declare and EOL date?

File: doc/source/ray-contribute/api-policy.rst [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+      - No
+      - No
+      - No
+      - Yes. The end-of-life date is a future Ray release version.
+      - No
+    * - Can this API be private (either inside the _internal module or has an underscore prefix)?
```
Comment (angelinalg) [inline doc/source/ray-contribute/api-policy.rst:56]:
```suggestion
    * - Can this API be private, either be inside the `_internal` module, or has an underscore prefix in the name?
```

File: doc/source/ray-contribute/api-policy.rst [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+      - No
+      - No
+      - No
+      - No
+      - No
+
+API Lifecycle Policy
```
Comment (angelinalg) [inline doc/source/ray-contribute/api-policy.rst:63]:
```suggestion
API lifecycle policy
```

File: doc/source/ray-contribute/api-policy.rst [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+~~~~~~~~~~~~~~~~~~~~
+Users have high expectations for certain exposure levels,
+so we need to be cautious when moving APIs between different
```
Comment (angelinalg) [inline doc/source/ray-contribute/api-policy.rst:66]:
```suggestion
so use cautioun when moving APIs between different
```

File: doc/source/ray-contribute/api-policy.rst [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+levels. Here is the policy for managing the API exposure lifecycle.
```
Comment (angelinalg) [inline doc/source/ray-contribute/api-policy.rst:67]:
```suggestion
levels. Below is the policy for managing the API exposure lifecycle.
```

File: doc/source/ray-contribute/api-policy.rst [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+    :widths: 20 16 16 16 16 16
+    :header-rows: 1
+
+    * - Policy/Exposure Level
+      - Stable Public API
+      - Beta Public API
+      - Alpha Public API
+      - Deprecated API
+      - Developer API
+    * - Can this API be promoted to a higher level without any warnings, heads up to users?
```
Comment (angelinalg) [inline doc/source/ray-contribute/api-policy.rst:79]:
```suggestion
    * - Can you promote this API to a higher level without any warnings or communication to users?
```

File: doc/source/ray-contribute/api-policy.rst [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+      - Yes
+      - Yes
+      - Yes
+      - No
+      - Yes
+    * - Can this API be demoted to a lower level? If so then how?
```
Comment (angelinalg) [inline doc/source/ray-contribute/api-policy.rst:85]:
```suggestion
    * - Can you demote this API to a lower level?
```

File: doc/source/ray-contribute/api-policy.rst [hunk 10]
```diff
@@ -0,0 +1,96 @@
...
+      - Can be demoted to Deprecated only. The API should emit warning messages and a deadline for deprecations in **6 months (or +25 ray minor versions)**.
```
Comment (angelinalg) [inline doc/source/ray-contribute/api-policy.rst:86]:
```suggestion
      - You can demote it to `Deprecated` only. The API should emit warning messages and you must set a deadline for deprecation in **6 months, or +25 Ray minor versions**.
```

File: doc/source/ray-contribute/api-policy.rst [hunk 11]
```diff
@@ -0,0 +1,96 @@
...
+      - Can be demoted to Deprecated only. The API should emit warning messages and a deadline for deprecations in **3 months (or +12 ray minor versions)**.
```
Comment (angelinalg) [inline doc/source/ray-contribute/api-policy.rst:87]:
```suggestion
      - You can demote it to `Deprecated` only. The API should emit warning messages and you must set a deadline for deprecations in **3 months or +12 Ray minor versions**.
```

File: doc/source/ray-contribute/api-policy.rst (hunk 10 above)
Comment (angelinalg) [reply doc/source/ray-contribute/api-policy.rst:86]: Add an inline link to what a Ray minor version is?

File: doc/source/ray-contribute/api-policy.rst [hunk 12]
```diff
@@ -0,0 +1,96 @@
...
+      - Breaking changes must be both allowed and expected in alpha components, and users must have no expectation of stability.
```
Comment (angelinalg) [inline doc/source/ray-contribute/api-policy.rst:88]:
```suggestion
      - Users must allow for and expect breaking changes in alpha components, and have no expectations of stability.
```

File: doc/source/ray-contribute/api-policy.rst [hunk 13]
```diff
@@ -0,0 +1,96 @@
...
+      - Yes
+      - No annotations mean it is a developer API by default
```
Comment (angelinalg) [inline doc/source/ray-contribute/api-policy.rst:90]:
```suggestion
      - No. The absence of annotations implies the Developer API level by default.
```

File: doc/source/ray-contribute/api-policy.rst [hunk 14]
```diff
@@ -0,0 +1,96 @@
...
+    * - Can this API remove or change it parameters?
```
Comment (angelinalg) [inline doc/source/ray-contribute/api-policy.rst:91]:
```suggestion
    * - Can you remove or change this API's parameters?
```

File: doc/source/ray-contribute/api-policy.rst [hunk 15]
```diff
@@ -0,0 +1,96 @@
...
+      - The API should emit warning messages and a deadline for the change in **6 months (or +25 ray minor versions)**. During the transition period, both the new and old parameters are supported
```
Comment (angelinalg) [inline doc/source/ray-contribute/api-policy.rst:92]:
```suggestion
      - Yes. The API should emit warning messages and you must set a deadline for the end-of-life of the original version that is **6 months or +25 Ray minor versions**. During the transition period, you must support both the new and old parameters.
```

File: doc/source/ray-contribute/api-policy.rst [hunk 16]
```diff
@@ -0,0 +1,96 @@
...
+      - The API should emit warning messages and a deadline for the change in **3 months (or +12 ray minor versions)**. During the transition period, both the new and old parameters are supported
```
Comment (angelinalg) [inline doc/source/ray-contribute/api-policy.rst:93]:
```suggestion
      - The API should emit warning messages and you must set a deadline for the change in **3 months or +12 Ray minor versions**. During the transition period, you must support both the new and old parameters.
```

File: doc/source/ray-contribute/api-policy.rst [hunk 17]
```diff
@@ -0,0 +1,96 @@
...
+      - Breaking changes must be both allowed and expected in alpha components, and users must have no expectation of stability.
```
Comment (angelinalg) [inline doc/source/ray-contribute/api-policy.rst:94]:
```suggestion
      - Users must allow for and expect breaking changes in alpha components, and must have no expectations of stability.
```

**Review by angelinalg — APPROVED** [review]
Thanks for creating this doc! 
Some style nits for consistency and clarification. Please consider using Vale. go/vale

**Comment by can-anyscale** [comment]
@angelinalg's comments

**Pull request merged**